Update CMake and use native CUDA language support

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62550
* __->__ #62445

PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc @malfet @seemethere

Differential Revision: [D31503350](https://our.internmc.facebook.com/intern/diff/D31503350)

peterbell10 committed:
WIP: Update CMake and use native CUDA language support

PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

malfet commented:
@peterbell10 this sounds great, but moving to 3.12 is a bit challenging, as cmake-3.10 is the only version available in ubuntu-18.04

malfet commented on CMakeLists.txt:
```diff
@@ -489,16 +503,14 @@ if(MSVC)
 
   # Try harder
-  list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "/w" "-w")
+  string(APPEND CMAKE_CUDA_FLAGS " -Xcompiler /w -w")
```
Hmm, can you please explain why do you want to go from `list` to `string`? `string(APPEND` is inherently bad if code is compiled in folder than have spaces in it.

peterbell10 replied:
A list is a string separated with semicolons, but `CMAKE_CUDA_FLAGS` expects a normal string so it will pass the semicolons into the command line invocation.

malfet commented:
When PR is finalized, please add an explanation why you are replacing `list(ADD` with `string(APPEND`, as this change was previously done to support compilation in folders with spaces
Also, some explanation would be good why all those extra `-Wno-this-or-that` are added to the code?

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "WIP: Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 commented on .jenkins/pytorch/build-asan.sh:
```diff
@@ -16,6 +16,9 @@ clang --version
 # symbolize=1: Gives us much better errors when things go wrong
 export ASAN_OPTIONS=detect_leaks=0:symbolize=1:detect_odr_violation=0
+if [ -n "$(which conda)" ]; then
+  export CMAKE_PREFIX_PATH=/opt/conda
```
For some reason the non-system cmake doesn't pick up the conda installed MKL without manually adding it to the prefix path.

peterbell10 commented on .jenkins/pytorch/win-test-helpers/build_pytorch.bat:
```diff
@@ -96,7 +96,7 @@ if "%USE_CUDA%"=="1" (
   curl -kL https://github.com/peterjc123/randomtemp-rust/releases/download/v0.3/randomtemp.exe --output %TMP_DIR_WIN%\bin\randomtemp.exe
   set RANDOMTEMP_EXECUTABLE=%TMP_DIR_WIN%\bin\nvcc.exe
-  set CUDA_NVCC_EXECUTABLE=%TMP_DIR_WIN%\bin\randomtemp.exe
+  set CUDA_NVCC_EXECUTABLE=%TMP_DIR%/bin/randomtemp.exe
```
I think this is issue might be cygwin related? It was interpreting the path as unix-style and so the backslashes were parsed as escape characters.

peterbell10 commented on cmake/Dependencies.cmake:
```diff
@@ -33,6 +33,50 @@ macro(enable_ubsan)
 endmacro()
 
+# ---[ CUDA
+if(USE_CUDA)
```
This code is the same, but moved to the top because `thread.cmake` uses `USE_CUDA` now.

peterbell10 reviewed (commented):
@malfet PTAL. Getting CI green was harder than expected but it's finally passing now.

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 commented:
@malfet ping

malfet commented on cmake/public/utils.cmake:
```diff
@@ -446,10 +459,18 @@ function(torch_compile_options libname)
+  target_compile_options(${libname} PRIVATE
+      $<$<COMPILE_LANGUAGE:CXX>:${private_compile_options}>)
+  if(USE_CUDA)
+    string(REPLACE ";" "," private_compile_options "${private_compile_options}")
```
Can you please add a short comment explaining that this converts list into string. 
And wouldn't `foreach` loop express the intent in a clearer way in this case?

peterbell10 replied:
I've added a comment to make the intention clear. I really do want a comma-separated list though, so a foreach would just end up as more code IMO.

malfet replied:
Sounds good, but I wonder if we need to check if private_compile_options is space-free?

peterbell10 commented on aten/src/ATen/native/sparse/cuda/SoftMax.cu:
```diff
@@ -392,8 +392,6 @@ void cuda_sparse_coo_softmax(
   }
 
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();
-  auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
-  auto policy = thrust::cuda::par(allocator).on(stream);
```
No, this PR propagates more C++ flags to nvcc so these unused variables fail the Werror build.

malfet replied:
I'm not an expert here, but aren't `policy` is something like a thread-guard/context manager (i.e. until this class is destroyed all thrust primitives are executed on the cuda stream specified by `stream` variable instead of a default one?
@ngimel can you confirm whether its the case of changes here are ok to remove?
P.S. Can you perhaps move this change into separate PR that I can land ASAP?

peterbell10 replied:
You can see from other parts of this file that the execution policy is passed explicitly to the function:
https://github.com/pytorch/pytorch/blob/e83d0a606ad3b72e7054ac56f04011226d6bca31/aten/src/ATen/native/sparse/cuda/SoftMax.cu#L244-L245

So, the fact it compiles without this variable proves that it's dead code. You can also see the function doesn't actually use thrust directly, instead it's all in `compute_pool_max` which defines it's own `policy`.

I've moved this change out into  #65539.

peterbell10 commented on c10/cuda/CMakeLists.txt:
```diff
@@ -49,9 +49,7 @@ if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
-target_link_libraries(c10_cuda PUBLIC c10)
-
-target_link_libraries(c10_cuda INTERFACE torch::cudart)
+target_link_libraries(c10_cuda PUBLIC c10 torch::cudart)
```
Because `c10_cuda` does need to linked against cudart and it's a quirk of FindCUDA that this wasn't necessary before (it was linked to cudart automatically).

peterbell10 commented on cmake/Dependencies.cmake:
```diff
@@ -1610,22 +1612,19 @@ if(NOT INTERN_BUILD_MOBILE)
     set(CMAKE_CXX_STANDARD 14)
   endif()
 
-  list(APPEND CUDA_NVCC_FLAGS ${TORCH_NVCC_FLAGS})
-  if(CMAKE_POSITION_INDEPENDENT_CODE AND NOT MSVC)
-    list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
-  endif()
-
```
Yes, I can look in `compile_commans.json` and see that `-Xcompiler=-fPIC` is in the command line even without this.

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 commented on cmake/public/cuda.cmake:
```diff
@@ -493,3 +504,7 @@ list(APPEND CUDA_NVCC_FLAGS "--expt-relaxed-constexpr")
+
+foreach(FLAG ${CUDA_NVCC_FLAGS})
+  string(APPEND CMAKE_CUDA_FLAGS " ${FLAG}")
+endforeach()
```
Done

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix CUDA_NVCC_EXECUTABLE on windows on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 commented:
@malfet any updates?

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]

peterbell10 committed:
Update on "Update CMake and use native CUDA language support"


PyTorch currently uses the old style of compiling CUDA in CMake which is just a
bunch of scripts in `FindCUDA.cmake`. Newer versions support CUDA natively as
a language just like C++ or C.

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

cc malfet seemethere

[ghstack-poisoned]
